torch/_dynamo/variables/lists.py

```diff
@@ -100,6 +101,49 @@ def call_method(
 
         return super(BaseListVariable, self).call_method(tx, name, args, kwargs)
 
+    def next_variables(self):
+        return unimplemented(f"{type(self)}.next_variables()")
+
+    @staticmethod
+    def generic_list_compare(left, tx, op, right, **options):
+        # Most list-like variables implement comparison ops the same way,
+        # so they can re-use this helper.
+        # There are quirks though, like how `tuple([2]) == torch.Size([2])`,
+        # but `tuple([2]) != list([2])`
+        if len(left.items) != len(right.items):
+            return ConstantVariable(False, **options)
+        if len(left.items) == 0:
+            return ConstantVariable(True, **options)
+
+        left_curr, left_rest = left.next_variables()
+        right_curr, right_rest = right.next_variables()
+        curr_compare = left_curr.compare(tx, op, right_curr, **options)
+        rest_compare = left_rest.compare(tx, op, right_rest)
+        if len(left.items) == 1:
+            return curr_compare
```

Comment:
Added a separate base case for 0 vs 1, so I could avoid adding handling for `operator.and_(DynamicShapeVariable, ConstantVariable)` (figured it was annoying to special case, and also is rare in user code)